pandas/tests/indexes/multi/test_get_set.py

```diff
@@ -345,6 +345,19 @@ def test_set_names_with_nlevel_1(inplace):
     tm.assert_index_equal(result, expected)
 
 
+def test_multi_set_names_pos_args_deprecation():
+    # GH#41485
+    idx = MultiIndex.from_product([["python", "cobra"], [2018, 2019]])
+
+    msg = (
+        "Starting with pandas version 2.0 all arguments of Index.set_names "
+        "except for the argument 'names' will be keyword-only"
+    )
+
+    with tm.assert_produces_warning(FutureWarning, match=msg):
+        idx.set_names(["kind", "year"], None)
```

Comment:
Can you also check the result? Again, I'm aware that I didn't do this in the example PR I opened, but then other reviewers suggested it - there's an example in #41511 . Same for the Series test

Reply:
No problem, added in the latest commit.